packages/rocketchat-lib/settings/server/startup.coffee

```diff
@@ -34,11 +34,14 @@ RocketChat.settings.add 'Accounts_OAuth_Twitter', false, { type: 'boolean', grou
 RocketChat.settings.add 'Accounts_OAuth_Twitter_id', '', { type: 'string', group: 'Accounts', section: 'Twitter' }
 RocketChat.settings.add 'Accounts_OAuth_Twitter_secret', '', { type: 'string', group: 'Accounts', section: 'Twitter' }
 
+RocketChat.settings.add 'Accounts_AllowUsernameChange', true, { type: 'boolean', group: 'Accounts', section: 'General', public: true }
```

Comment:
Yes @graywolf336. Client needs to know if username change is allowed or not, so it can gray out username field or whatever it does to inform the user he/she cannot change the username before actually submitting a username change to the server.